src/overlays/actors/ovl_Bg_Hidan_Hamstep/z_bg_hidan_hamstep.c

```diff
@@ -25,33 +58,343 @@ const ActorInit Bg_Hidan_Hamstep_InitVars = {
     (ActorFunc)BgHidanHamstep_Update,
     (ActorFunc)BgHidanHamstep_Draw,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Hidan_Hamstep/func_80888040.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Hidan_Hamstep/func_8088805C.s")
+static BgHidanHamstepActionFunc sActionFuncs[] = {
+    func_808887C4, func_80888860, func_808889B8, func_80888A58, BgHidanHamstep_DoNothing,
+};
+
+static InitChainEntry sInitChain[] = {
+    ICHAIN_VEC3F_DIV1000(scale, 100, ICHAIN_STOP),
+};
+
+static f32 sEffectPositions[][2] = {
+    { -100.0f, 40.0f },  { 100.0f, 40.0f }, { -100.0f, 0.0f },   { 100.0f, 0.0f },
+    { -100.0f, -40.0f }, { 100.0f, 40.0f }, { -100.0f, -80.0f }, { -50.0f, -80.0f },
+    { 0.0f, -80.0f },    { 50.0f, -80.0f }, { 100.0f, -80.0f },
+};
+
+void BgHidanHamstep_SetupAction(BgHidanHamstep* this, s32 actionState) {
+    this->actionState = actionState;
+    this->actionFunc = sActionFuncs[actionState];
```

Comment:
We generally just call this kind of field `action`, so I suggest using that here
```suggestion
void BgHidanHamstep_SetupAction(BgHidanHamstep* this, s32 action) {
    this->action = action;
    this->actionFunc = sActionFuncs[action];
```